### ISSUE TITLE
Uncaught Error: Patch does not apply

### ISSUE THREAD

**Issue by khan-c** [atom/atom#16302]
[Enter steps to reproduce:]

1. type something

**Atom**: 1.22.1 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.12.6
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: Patch does not apply

```
At /Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/helpers.js:68

Error: Patch does not apply
    at exports.patchFromChanges (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/helpers.js:68:17)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:1692:68)
    at later (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/helpers.js:28:13)
```

### Commands

```
     -1:06.2.0 editor:delete-to-beginning-of-line (input.hidden-input)
     -1:05.0 core:backspace (input.hidden-input)
     -1:02.6.0 editor:toggle-line-comments (input.hidden-input)
 26x -1:01.4.0 core:undo (input.hidden-input)
     -0:46.7.0 core:save (input.hidden-input)
     -0:44.7.0 editor:toggle-line-comments (input.hidden-input)
     -0:43.1.0 editor:delete-to-beginning-of-line (input.hidden-input)
     -0:42.7.0 snippets:next-tab-stop (input.hidden-input)
     -0:42.7.0 snippets:expand (input.hidden-input)
     -0:42.7.0 editor:indent (input.hidden-input)
     -0:37.2.0 core:move-right (input.hidden-input)
     -0:36.4.0 editor:newline (input.hidden-input)
     -0:34.1.0 core:move-right (input.hidden-input)
     -0:33.8.0 editor:move-to-end-of-screen-line (input.hidden-input)
  2x -0:33.1.0 core:save (input.hidden-input)
     -0:16.9.0 editor:delete-to-beginning-of-line (input.hidden-input)
```

### Non-Core Packages

```
busy-signal 1.4.3 
color-picker 2.2.5 
file-icons 2.1.14 
highlight-selected 0.13.1 
intentions 1.1.5 
linter 2.2.0 
linter-eslint 8.4.0 
linter-rubocop 2.2.0 
linter-ui-default 1.6.10 
pigments 0.40.2 
react 0.17.0 
rspec 0.4.0 
toggle-quotes 1.0.1 
```

**Comment by grimmlink** [comment]
Same probleme here, many error popups..

**Atom**: 1.22.1 ia32
**Electron**: 1.6.15
**OS**: Windows 10
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Patch does not apply

```
At C:\Users\grimm\AppData\Local\atom\app-1.22.1\resources\app\node_modules\text-buffer\lib\helpers.js:68

Error: Patch does not apply
    at exports.patchFromChanges (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/helpers.js:68:17)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/text-buffer.js:1692:68)
    at later (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/helpers.js:28:13)
```

### Commands

```
     -0:48 pane:move-active-item-to-top-of-stack (input.hidden-input)
     -0:45 editor:move-to-first-character-of-line (input.hidden-input)
  2x -0:44.5.0 editor:add-selection-below (input.hidden-input)
     -0:39.7.0 core:delete (input.hidden-input)
     -0:39.3.0 autocomplete-plus:cancel (input.hidden-input)
     -0:38.2.0 editor:select-to-beginning-of-word (input.hidden-input)
     -0:38 core:delete (input.hidden-input)
     -0:37.4.0 core:save (input.hidden-input)
     -0:23 editor:consolidate-selections (input.hidden-input)
     -0:23 core:cancel (input.hidden-input)
     -0:19.5.0 core:backspace (input.hidden-input)
  2x -0:19 core:move-up (input.hidden-input)
     -0:18 editor:newline (input.hidden-input)
     -0:17.3.0 core:move-up (input.hidden-input)
  4x -0:17.1.0 core:delete (input.hidden-input)
     -0:15.4.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
highlight-selected 0.13.1 
php-twig 4.0.0 
Remote-FTP 2.0.0 
Sublime-Style-Column-Selection 1.7.4 
```

**Comment by IlonaF** [comment]
Same problem... any ideas as to what is causing it?

**Atom**: 1.22.1 ia32
**Electron**: 1.6.15
**OS**: Microsoft Windows 10 Home
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Patch does not apply

```
At C:\Users\Yellonqa\AppData\Local\atom\app-1.22.1\resources\app\node_modules\text-buffer\lib\helpers.js:68

Error: Patch does not apply
    at exports.patchFromChanges (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/helpers.js:68:17)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/text-buffer.js:1692:68)
    at later (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/helpers.js:28:13)
```

### Commands

```
  3x -2:16.3.0 core:backspace (input.hidden-input)
     -2:08.0 core:save (input.hidden-input)
     -1:35 core:move-down (input.hidden-input)
     -1:34.1.0 core:move-up (input.hidden-input)
     -1:33.7.0 core:move-left (input.hidden-input)
     -1:31.3.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
compare-files 0.8.1 
```

**Comment by khan-c** [comment]
No idea what's causing it. I quit and reopened Atom and didn't get the error anymore. The good old "turn it off and on again" fix...

**Comment by rsese** [comment]
Thanks for the reports :+1:  This issue is being investigated here so you can subscribe for updates if you'd like:

https://github.com/atom/atom/issues/15827

If anyone has details that might help reproducing the error, please share in https://github.com/atom/atom/issues/15827.

---

We noticed that this is a duplicate of https://github.com/atom/atom/issues/15827. You may want to subscribe there for updates.

Because we treat our issues list as [the Atom team's backlog](https://en.wikipedia.org/wiki/Scrum_(software_development)#Product_backlog), we close duplicates to focus our work and not have to touch the same chunk of code for the same reason multiple times. This is also why we may mark something as duplicate that isn't an exact duplicate but is closely related.

For information on how to use GitHub's search feature to find out if something is a duplicate before filing, see [the How Can I Contribute? section of the Atom CONTRIBUTING guide](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#how-can-i-contribute).

**Comment by alissonpelizaro** [comment]
Hello ... I was noticing that bug boxes usually opened when I start a php tag. What do you think about it?

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.22.1 ia32
**Electron**: 1.6.15
**OS**: Windows 10
**Thrown From**: Atom Core


### Stack Trace

Uncaught Error: Patch does not apply

```
At C:\Users\Alisson\AppData\Local\atom\app-1.22.1\resources\app\node_modules\text-buffer\lib\helpers.js:68

Error: Patch does not apply
    at exports.patchFromChanges (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/helpers.js:68:17)
    at TextBuffer.module.exports.TextBuffer.emitDidStopChangingEvent (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/text-buffer.js:1692:68)
    at later (~/AppData/Local/atom/app-1.22.1/resources/app/node_modules/text-buffer/lib/helpers.js:28:13)
```

### Commands

```
  7x -3:50.8.0 core:backspace (input.hidden-input)
  2x -3:36.5.0 core:save (input.hidden-input)
  6x -2:57.6.0 core:backspace (input.hidden-input)
     -2:53.5.0 autocomplete-plus:confirm (input.hidden-input)
     -2:53.4.0 snippets:expand (atom-text-editor.editor.is-focused)
     -2:51.7.0 editor:newline (input.hidden-input)
  2x -2:49 core:backspace (input.hidden-input)
     -2:31.9.0 core:move-up (input.hidden-input)
 13x -2:11.4.0 core:delete (input.hidden-input)
     -2:07.7.0 editor:move-to-end-of-screen-line (input.hidden-input)
 12x -2:06.3.0 core:move-left (input.hidden-input)
  5x -2:05.0 core:move-right (input.hidden-input)
     -2:02.5.0 core:move-down (input.hidden-input)
  5x -1:49.1.0 core:backspace (input.hidden-input)
     -1:41.5.0 autocomplete-plus:confirm (input.hidden-input)
     -1:41.5.0 snippets:expand (atom-text-editor.editor.is-focused)
```

### Non-Core Packages

```

```

**Comment by rsese** [comment]
@alissonpelizaro - it looks like https://github.com/atom/atom/issues/15827 has been closed and the fix should go out in 1.25 next month /cc https://github.com/atom/atom/issues/15827#issuecomment-355138140.

Is the error still continually showing up for you with an opening PHP tag?
